tools/xibuild/Main.cs

```diff
@@ -115,23 +115,27 @@ static int RunTool (string toolPath, string combinedArgs, string baseConfigFile)
 		{
 			var tmpMSBuildExePathForConfig = Path.GetTempFileName ();
 			var configFilePath = tmpMSBuildExePathForConfig + ".config";
-
-			GenerateAppConfig (configFilePath, baseConfigFile, out string MSBuildSdksPath);
-
-			var psi = new ProcessStartInfo {
-				FileName = toolPath,
-				Arguments = combinedArgs,
-				UseShellExecute = false,
-			};
-			// Required so that msbuild can read the correct config file
-			psi.EnvironmentVariables ["MSBUILD_EXE_PATH"] = tmpMSBuildExePathForConfig;
-			// MSBuildSDKsPath only works via an env var
-			psi.EnvironmentVariables ["MSBuildSDKsPath"] = MSBuildSdksPath;
-
-			var p = Process.Start (psi);
-
-			p.WaitForExit ();
-			return p.ExitCode;
+			try {
+				GenerateAppConfig (configFilePath, baseConfigFile, out string MSBuildSdksPath);
+
+				var psi = new ProcessStartInfo {
+					FileName = toolPath,
+					Arguments = combinedArgs,
+					UseShellExecute = false,
+				};
+				// Required so that msbuild can read the correct config file
+				psi.EnvironmentVariables ["MSBUILD_EXE_PATH"] = tmpMSBuildExePathForConfig;
+				// MSBuildSDKsPath only works via an env var
+				psi.EnvironmentVariables ["MSBuildSDKsPath"] = MSBuildSdksPath;
+
+				var p = Process.Start (psi);
+
+				p.WaitForExit ();
+				return p.ExitCode;
+			} finally {
+				File.Delete (tmpMSBuildExePathForConfig);
+				File.Delete (configFilePath);
```

Comment:
Approving, but I would add a check to make sure that paths are there before deleting.. To avoid any possible exceptions.